entrypoints_cache.py

```diff
@@ -0,0 +1,168 @@
+"""Use a cache layer in front of entry point scanning."""
+
+import errno
+import glob
+import hashlib
+import itertools
+import json
+import logging
+import os
+import os.path
+import struct
+import sys
+
+import entrypoints
+
+NoSuchEntryPoint = entrypoints.NoSuchEntryPoint
+BadEntryPoint = entrypoints.BadEntryPoint
+EntryPoint = entrypoints.EntryPoint
+
+
+log = logging.getLogger('entrypoints_cache')
+
+
+def _get_cache_dir():
+    """Locate a platform-appropriate cache directory to use.
+
+    Does not ensure that the cache directory exists.
+    """
+    # Linux, Unix, AIX, etc.
+    if os.name == 'posix' and sys.platform != 'darwin':
+        # use ~/.cache if empty OR not set
+        return os.environ.get("XDG_CACHE_HOME", None) \
```

Comment:
The code should use a `python-entrypoints` subdir even with XDG_CACHE_HOME set.

Reply:
Great point, done.